ios/Exponent/Kernel/Services/EXUserNotificationManager.m

```diff
@@ -0,0 +1,58 @@
+// Copyright 2015-present 650 Industries. All rights reserved.
+
+#import "EXUserNotificationManager.h"
+#import "EXKernel.h"
+#import "EXRemoteNotificationManager.h"
+#import "EXEnvironment.h"
+
+@interface EXUserNotificationManager ()
+
+@property (nonatomic, strong) NSMutableDictionary<NSString *, EXPendingNotification *> *pendingNotifications;
+
+@end
+
+@implementation EXUserNotificationManager
+
+- (instancetype)init
+{
+  if (self = [super init]) {
+    _pendingNotifications = [NSMutableDictionary new];
+  }
+  return self;
+}
+
+- (EXPendingNotification *)initialNotificationForExperience:(NSString *)experienceId
+{
+  return _pendingNotifications[experienceId];
+}
+
+# pragma mark - UNUserNotificationCenterDelegate
+
+- (void)userNotificationCenter:(UNUserNotificationCenter *)center didReceiveNotificationResponse:(UNNotificationResponse *)response withCompletionHandler:(void (^)(void))completionHandler
```

Comment:
It looks like this method is called also when the user dismisses a notification from the iOS notification center UI. We probably don't want to send an event to JS in that case.

Reply:
Have you experienced it or read in the docs? From my experience and understanding we would needed to add a custom category with [`UNNotificationCategoryOptionCustomDismissAction `](https://developer.apple.com/documentation/usernotifications/unnotificationcategoryoptions/unnotificationcategoryoptioncustomdismissaction?language=objc) option for this method to be called when the user dismisses the notification, but I'll reassure that during QA of this PR.

Reply:
Confirmed! This method won't be called when a user dismisses notification with current configuration.